docs/AssetFormats.md

```diff
@@ -0,0 +1,106 @@
+# Asset Formats
+This document describes the data layouts for NcEngine assets and asset file types.
+
+## Contents
+- [Nc Asset](#nc-asset)
+  - [.nca File Format](#nca-file-format)
+- [Nc Asset Package](#nc-asset-package)
+  - [.ncp File Format](#ncp-file-format)
+  - [LUT Entry Format](#lut-entry-format)
+- [Blob Formats](#blob-formats)
+    - [AudioClip](#audioclip-blob-format)
+    - [ConcaveCollider](#concavecollider-blob-format)
+    - [Cubemap](#cubemap-blob-format)
+    - [HullCollider](#hullcollider-blob-format)
+    - [Mesh](#mesh-blob-format)
+    - [Shader](#shader-blob-format)
+    - [Texture](#texture-blob-format)
+
+## Nc Asset
+-----------
+An asset can exist on its own in a .nca file, or as an entry in a .ncp package. The data layout is identical in either case. Each nca consists of a header that is common between all asset types and
+an asset blob, the layout of which is determined by the asset type.
+
+### .nca File Format
+| Name         | Type    | Size | Note |
+|--------------|---------|------|------
+| magic number | string  | 4    | non-null-terminated string identifying the asset type
```

Comment:
can you change to identifier, uid, or something else? 'magic number' is actually a negative thing - it's a term used to represent an anti-pattern

Reply:
Per Wikipedia: 
The term magic number or magic constant refers to the [anti-pattern](https://en.wikipedia.org/wiki/Anti-pattern) of using numbers directly in source code.